tealium/collectors/momentsapi/MomentsAPIHTTPError.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  MomentsAPIHTTPError.swift
+//  tealium-swift
+//
+//  Copyright © 2024 Tealium, Inc. All rights reserved.
+//
+
+import Foundation
+
+enum MomentsAPIHTTPError: Int, Error, LocalizedError {
+    case success = 200
```

Comment:
I know that this is checked outside, but why is `success` part of an Error enum?

Additionally, should the server crash and return a 500 response, or support other 400 errors, should we also accept other status codes as errors? Otherwise they will all return as `URLError(.cannotDecodeRawData)` again.

I was thinking of something like this:
```swift
enum MomentsAPIHTTPError: RawRepresentable, Error, LocalizedError {
    typealias RawValue = Int

    case badRequest
    case forbidden
    case notFound
    case unsuccessful(_ statusCode: Int)

    init?(rawValue: Int) {
        switch rawValue {
        case 200..<300:
            return nil
        case 400:
            self = .badRequest
        case 403:
            self = .forbidden
        case 404:
            self = .notFound
        default:
            self = .unsuccessful(rawValue)
        }
    }
    
    var rawValue: Int {
        switch self {
        case .badRequest:
            return 400
        case .forbidden:
            return 403
        case .notFound:
            return 404
        case .unsuccessful(let code):
            return code
        }
    }
...
}
```